apps/gbfs/lib/page_cache.ex

```diff
@@ -0,0 +1,87 @@
+defmodule PageCache do
+  @moduledoc """
+  This module provides the ability to cache a HTTP response (in RAM, currently using Cachex).
+
+  It is implemented as a Plug, so that you can plug it in any given router.
+
+  In case of technical error (e.g. cache not available), the query should still be honored,
+  but without caching.
+
+  Improvements that would make sense:
+  - let the caller build the cache key
+  - let the caller decide when not to cache, or specify dynamic ttl
+  - let the caller handle errors
+  """
+
+  import Plug.Conn
+  require Logger
+
+  def init(options) do
+    ttl_seconds = options |> Keyword.fetch!(:ttl_seconds)
+    options |> Keyword.put(:ttl, :timer.seconds(ttl_seconds))
+  end
+
+  defmodule CacheEntry do
+    @moduledoc """
+    The CacheEntry contains what is serialized in the cache currently.
+    """
+    defstruct [:body, :content_type]
+  end
+
+  def build_cache_key(request_path) do
+    ["page", request_path] |> Enum.join(":")
+  end
+
+  def call(conn, options) do
+    page_cache_key = build_cache_key(conn.request_path)
+
+    options
+    |> Keyword.fetch!(:cache_name)
+    |> Cachex.get(page_cache_key)
+    |> case do
+      {:ok, nil} -> handle_miss(conn, page_cache_key, options)
+      {:ok, value} -> handle_hit(conn, page_cache_key, value)
+      {:error, error} -> handle_error(conn, error)
+    end
+  end
+
+  def handle_miss(conn, page_cache_key, options) do
+    Logger.info("Cache miss for key #{page_cache_key}")
```

Comment:
No strong opinion either ; I just wanted to be able to see that in production, because we do not have that many queries yet. In the future I'll prefer to have metrics with cache hit / miss ratio etc.

Reply:
After checking, only info logs are available on production, so we'll keep it as info for now.